### PR TITLE
fix: Investigate NavigationUIComponent

### LINKED ISSUE #7038: Investigate NavigationUIComponent
Component cleanup.

### PR DESCRIPTION
**Pull request by Fonsu**
Closes #7038

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts [hunk 1]
```diff
@@ -185,8 +185,4 @@ export class NavigationUIComponent implements OnDestroy {
 
     this.isOpen = this.openNodes.length > 0;
   }
-
```
Comment (kacperknapik) [inline projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts:196]: Was this function not used?

File: projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/tabbing-order.e2e-spec.ts [hunk 2]
```diff
@@ -99,7 +96,6 @@ describe("Tabbing order - tests don't require user to be logged in", () => {
 
   describe('Header Sub Categories - Desktop', () => {
     context('Brands', () => {
-      // TODO: Test currently fails because navigation-ui works incorrectly
```
Comment (kacperknapik) [inline projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/tabbing-order.e2e-spec.ts]: Is it fixed already? If not, please leave this comment here

File: projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/tabbing-order.e2e-spec.ts [hunk 3]
```diff
@@ -115,7 +111,6 @@ describe("Tabbing order - tests don't require user to be logged in", () => {
     });
 
     context('Accessories', () => {
-      // TODO: This test fails with the current navigation-ui implementation for unknown reasons.
```
Comment (kacperknapik) [inline projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/tabbing-order.e2e-spec.ts]: Same as above

**Review by kacperknapik — CHANGES_REQUESTED** [review]

File: projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/tabbing-order.e2e-spec.ts (hunk 2 above)
Comment (Fonsu) [reply projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/tabbing-order.e2e-spec.ts]: Yes, that particular e2e desktop test is now fixed.

File: projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/tabbing-order.e2e-spec.ts (hunk 3 above)
Comment (Fonsu) [reply projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/tabbing-order.e2e-spec.ts]: Yes, that particular e2e desktop test is now fixed.

File: projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts (hunk 1 above)
Comment (Fonsu) [reply projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts:196]: It was not used.

**Review by kacperknapik — APPROVED** [review]

File: projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts [hunk 4]
```diff
@@ -185,8 +185,4 @@ export class NavigationUIComponent implements OnDestroy {
...
-  isTabbable(node: any) {
```
Comment (kacperknapik) [inline projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts:197]: This removal needs to be documented.

**Review by kacperknapik — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts (hunk 4 above)
Comment (Fonsu) [reply projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts:197]: done

**Review by kacperknapik — DISMISSED** [review]

**Review by kacperknapik — APPROVED** [review]

**Pull request merged**